csrc/multidevice/utils.cpp

```diff
@@ -140,6 +140,10 @@ int64_t numDeviceDims(const TensorView* tv) {
 bool haveDifferentShardings(
     const TensorView* producer,
     const TensorView* consumer) {
+  // cpu scalars are not required to have a mesh
+  if (producer->isCpuScalar() || consumer->isCpuScalar()) {
```

Comment:
Is it possible to have `producer` being a sharded tensor while `consumer` being a CPU scalar? What would be the expected behavior in that case?

Reply:
Hm I don't think we have any exprs that produce a CPU scalar, so I'll just check if the producer is a CPU scalar

Reply:
Doesn't SpdaFwdOp produce CPU scalars for seed and offset?

Reply:
> Hm I don't think we have any exprs that produce a CPU scalar, so I'll just check if the producer is a CPU scalar

`SdpaFwdOp` has philox seed and philox offset as CPU scalar consumers.

Reply:
ahh yes, you're right - changing back to the original